core/src/main/scala/kafka/server/BrokerLifecycleManager.scala

```diff
@@ -44,6 +44,13 @@ import scala.concurrent.duration._
  * Explicit broker state transitions are performed by co-ordinating
  * with the controller through the heartbeats.
  *
+ * Expected state transitions when starting up are:
+ * SHUTDOWN => REGISTERING => FENCED [=> RECOVERING_FROM_UNCLEAN_SHUTDOWN] => RUNNING
+ *
+ * We potentially transition from RUNNING to FENCED and back to RUNNING
+ *
+ * We eventually shutdown:
+ * RUNNING [=> PENDING_CONTROLLED_SHUTDOWN] => SHUTDOWN
```

Comment:
Should this say "NOT_RUNNING" instead of "SHUTDOWN" to stay consistent with the BrokerState?